src/units.h

```diff
@@ -487,4 +490,61 @@ inline constexpr units::quantity<double, units::energy_in_millijoule_tag> operat
     return units::from_kilojoule( v );
 }
 
+template<typename T>
+inline T read_from_json_string( JsonIn &jsin, const std::vector<std::pair<std::string, T>> &units )
```

Comment:
This probably shouldn't be marked `inline` (it doesn't have to be, even though it's in a header, because it's a function template).

It's not a big deal, because compilers often don't respect `inline` anyway, but it's a big and complicated enough function that I feel it ought not to be `inline`.

Reply:
See #32223.